docs/adr/ADR-005-User_attribution_immutable_state_records.md

```diff
@@ -0,0 +1,197 @@
+- Application functionality should not depend on data that might be removed or relocated
+- Clear separation of concerns: audit data serves compliance/security needs, not functional needs
+
+This leaves us with a question: how do we handle user attribution for functional purposes?
+
+Within our application, we have two types of data:
+
+1. **Mutable records**: Records that can be updated over time (e.g., a `Participant` record where or some attributes may be updadted on the same record)
+2. **Immutable state records**: Records that represent a point-in-time state transition and are never modified once created (e.g., `AppointmentStatus` records that track the progression of an appointment through states like CONFIRMED, CHECKED_IN, SCREENED)
+
+For mutable records, simply adding a `user_id` field would be problematic. The field would only show the last user who modified the record, losing all historical context about who made which changes.
+
+For immutable state records, the situation is different. These records naturally preserve history because they are never modified. Each status transition creates a new record, building a complete timeline. Adding a `user_id` field to these records provides exactly what we need for functional attribution without additional complexity.
+
+Currently, we have `AppointmentStatus` as our first immutable state record model (We also have `ClinicStatus` but no understanding of attribution requirements of that as yet). An appointment progresses through states (CONFIRMED, CHECKED_IN, SCREENED, etc.), and each transition creates a new `AppointmentStatus` record. In the future, we anticipate similar patterns for other workflows e.g.:
+
+- Image reading workflow: States like AwaitingFirstRead, FirstReadCompleted, AwaitingSecondRead, SecondReadCompleted
+- Results workflow: States tracking the progression of screening results
+- Referral workflow: States tracking referral processing
+
+These future workflows would benefit from the same attribution pattern.
+
+## Decision
+
+We will add a `user_id` foreign key field to immutable state record models, starting with `AppointmentStatus`. This field will reference our `User` model and track which user created each state transition.
+
+### Key characteristics of this approach:
+
+- **Only immutable records**: User attribution via direct foreign key is only added to models that represent immutable state transitions, never to mutable records
+- **Pattern for future workflows**: This pattern will be applied to future state-based models (e.g., image reading states, results states)
+- **Functional, not audit**: This data is for functional purposes (displaying in UI, business rules) and is separate from audit logging
+- **Required field**: The `user_id` field will be required (not nullable) to ensure attribution is always captured
+- **System user support**: For automated state transitions, we will use a designated system user account rather than allowing null values
```

Comment:
This is described as "we will" but it is not part of the decision above, and below it says "we may"

It can be problematic to decide too much before implementing so maybe we can drop this bullet point and leave in the mention in the consequences section?